### ISSUE TITLE
Airstrike throwable/special

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#167]
Definitely needed. This game desperately lacks a solution for flushing people from behind walls. (Grenades are better than nothing, but they still suck at it.) An Airstrike (or Orbital Strike, or some other nonsense you come up with - doesn't matter, what matters is the gameplay) which would allow a player to designate some small (5x5) area of the map as about to be blasted would be a very useful addition.

Importantly, there should be enough time (5 seconds?) for the enemy to move out of the way of the incoming airstrike. This is not a tool for killing; this is a utility for countering cover. Furthermore, even if we add it to the throwable slot, it should basically ignore walls (i.e. i should be able to call the Airstrike onto an enemy that is behind a wall), else that defeats its purpose.

**Comment by JacKDot2** [comment]
me when the enemy team spam them like columns: (they are trying to start operation Rolling Thunder)

**Comment by deferred-impact** [comment]
Apparently this isn't clear and needs to be said explicitly. You cannot launch the airstrike at any point on the map. You can only use it on a spot you can directly see. It's basically a teleporting grenade which goes through walls.

For mobile controls: the circle appears on top of you and slowly moves away from you as you hold the firing joystick. Once you release the joystick the circle is placed.

**Comment by ghost** [comment]
Eh, Its like a timed c4. 

I'd rather use frag, or even c4, rather than use an airstrike in a tough situation against multiple people. 
As for 'zone restriction', the weapon barely has any range, so I doubt it's going to be effective in a gamemode like hardpoint or payload. And even if it is effective, it'll turn into a weapon that is just as annoying as flashbangs or smoke grenades, because people will just spam them into the control point, or in the payload. 

As for the weapon going through walls, I don't thinks that's gonna be very effective, or skillful. If you can throw it through walls, people would simply be able to use a frag grenade, without having to worry about cooking it, through a wall, and not having to worry about protection at all. 
The other team is advancing the payload?
No worries, i'll grab an ammo pack, hide behind a wall, and throw my airstrikes, which will clear out the zone, and I won't even get attacked.

And to make the whole thing more idiotic, the dev decided to make it depend on your killstreak lmfao.

**Comment by ghost** [comment]
Honestly, I love the concept of a zone limiting explosives, something that will force people to move out of a zone, but I don't think that this will do, because it's range is only so much. 
I might work for payload, but then again, you'll find people just spamming them, while having cover from a wall, which will  make the weapon annoying and overpowered. 

Earlier, as deffered had mentioned in the discord, the airstrike was only to have as much radius as a frag, 

![Screenshot 2023-10-19 8 51 01 PM](https://github.com/deferred-impact/battledudes-issue-tracker/assets/148166198/0dfdcddc-ef50-443c-99ab-dbf1290b99ef)
 
 which isn't quite large. But now, reading the description, seeing that the decided range is 5x5, I think it might be suitable for Hardpoint and Control points. However, the issue of players spamming it, and being able to conveniently hide behind cover while using it, remains an issue.

**Comment by deferred-impact** [comment]
> Eh, Its like a timed c4.

No, you cannot reduce this concept to a "timed C4". First of all, a "timed C4" is called a grenade. Secondly, grenades require line of sight. The entire point about this idea here is that line of sight is not needed. If someone is hiding behind a wall, you cannot grenade them. But, you can call an airstrike on them.

> I'd rather use frag, or even c4, rather than use an airstrike in a tough situation against multiple people.

Wonderful! Then use those. Nobody is forcing you to use the airstrike. If you fail to see the benefit of being able to flush people out of cover, then stick to the more conventional throwables.

> As for 'zone restriction', the weapon barely has any range, so I doubt it's going to be effective in a gamemode like hardpoint or payload. And even if it is effective, it'll turn into a weapon that is just as annoying as flashbangs or smoke grenades, because people will just spam them into the control point, or in the payload.

Every throwable is annoying in the control point gamemodes, it's pretty much tied to the label.

If you're that upset about airstrikes on Hardpoint, then a simple condition of "you cannot call an airstrike inside of a point" solves that completely.

> As for the weapon going through walls, I don't thinks that's gonna be very effective, or skillful. If you can throw it through walls, people would simply be able to use a frag grenade, without having to worry about cooking it, through a wall, and not having to worry about protection at all.

A frag grenade explodes faster. With the airstrike always having 5 seconds between the activation and the explosion, it will be easy to avoid. Saying that you "won't need to worry about protection" is also quite silly - the enemy will see who called the airstrike on them, and will gladly pursue you and teach you why you shouldn't do that.

> The other team is advancing the payload?
No worries, i'll grab an ammo pack, hide behind a wall, and throw my airstrikes, which will clear out the zone, and I won't even get attacked.

As i said on discord, do that, and you'll find the entire enemy team triangulating all their troops on your position

> Honestly, I love the concept of a zone limiting explosives, something that will force people to move out of a zone, but I don't think that this will do, because it's range is only so much.

Well the range is up to discussion and subsequent balancing. I can't randomly pick an arbitrary number and immediately pick the best one. To me a frag grenade explosion seems sufficiently large, but actual in-game testing would be needed

> However, the issue of players spamming it, and being able to conveniently hide behind cover while using it, remains an issue.

Again. The issue of players spamming throwables is a property of all throwables. And it can be fixed, e.g. #77

**Comment by ghost** [comment]
> If someone is hiding behind a wall, you cannot grenade them. But, you can call an airstrike on them.

Yes, so essentially a grenade that can be used behind walls, which is too overpowered. Also, please explain to me the use of this outside of Hardpoint, CP, or Payload, what would be the point of me trying to attack someone through a wall? It most likely won't kill them, they'll just move away, which would be stupid. 

> Every throwable is annoying in the control point gamemodes, it's pretty much tied to the label.

Doesn't mean that somehow makes them fair weapons, they are annoying for the gamemode, which means they must be balanced (maybe by buffing trophy) but that is for another issue. 

> A frag grenade explodes faster. With the airstrike always having 5 seconds between the activation and the explosion, it will be easy to avoid. 

Ok...? So like its just a slow grenade going through walls, which I again, do not understand the purpose of:
A. It hardly has any more range than frag, so its barely 'zone restricting'
B. It has a 5 second cooldown, so you can't exactly kill with it
So it is supposed to restrict a location but the same time, it lacks the necessities to do so. 

> "won't need to worry about protection" is also quite silly - the enemy will see who called the airstrike on them, and will gladly pursue you and teach you why you shouldn't do that.

I could be hiding inside a building, and could very easily go unseen. I could hide behind a placeable wall, or an object, while attacking them, and they might not even realize, because I could be hiding behind a tree. 
I could also just kill them if the try to attack me. 

> As i said on discord, do that, and you'll find the entire enemy team triangulating all their troops on your position

They get busty chasing me, while my team promptly takes control of the payload and pushes the cart back in the absence of the enemy team. It's a sacrifice I'm willing to pay. 

> Well the range is up to discussion and subsequent balancing. I can't randomly pick an arbitrary number and immediately pick the best one. To me a frag grenade explosion seems sufficiently large, but actual in-game testing would be needed

Yes, I suggest you try actual in-game testing, hence we can learn how effective the weapon is in game, and just exactly how similar it is to frag. 


> Again. The issue of players spamming throwables is a property of all throwables. And it can be fixed, e.g. https://github.com/deferred-impact/battledudes-issue-tracker/issues/77 

Well that depends solely on the implementation of issue 77.

**Comment by deferred-impact** [comment]
> Yes, so essentially a grenade that can be used behind walls, which is too overpowered. 

How is it overpowered? You have 5 seconds to leave the area that is about to be blasted. This airstrike ability would literally be less powerful than frag grenades. It's not supposed to get kills, and it would rarely get any

> Also, please explain to me the use of this outside of Hardpoint, CP, or Payload, what would be the point of me trying to attack someone through a wall?

I engage someone in a fight. I hit them once with my Flintlock shot. They immediately hide behind a wall. Now, i no longer have any good reliable way to hit the second, finishing shot, before the regen kicks in. If i go directly towards them, i end up in a close range fight, which is suboptimal for my playstyle. If i try to find a different angle while maintaining the range, the opponent will just keep running away while hugging the wall. And for all of this i have about 10 seconds total, as after that their regen will kick in, resetting the fight.

Instead i could call an airstrike on them, forcing them to move in a specific way, perhaps forcing them to leave their cover and walk into my line of fire.

> It most likely won't kill them, they'll just move away, which would be stupid.

You literally just said it's an overpowered weapon, now you're saying it won't kill people. Both can't be true at the same time.

The point is to make them move, yes. That is the entire point of this suggestion. An ability that lets players flush out other players from behind cover.

>  A. It hardly has any more range than frag, so its barely 'zone restricting'

So increase the range then. As i said multiple times, the range is an educated guess. It would need to be updated after further testing.

>  So it is supposed to restrict a location but the same time, it lacks the necessities to do so.

Then give it the necessities...

> I could be hiding inside a building, and could very easily go unseen. I could hide behind a placeable wall, or an object, while attacking them, and they might not even realize, because I could be hiding behind a tree.

And the problem with this is what? Let me tell you, i'd rather deal with some idiot trying to airstrike me (and failing, as per the 5 second delay) from within a roof, than deal with some idiot throwing a cooked grenade from under a roof, straight at me, which instantly explodes on me and leaves me no chance to retaliate. 

The airstrike is strictly weaker than the frag grenade, so i fail to see why you're complaining.

> I could also just kill them if the try to attack me.

Oh no, Yoyo is threatening us with the idea of him playing the game the way it's supposed to be played! If you kill them, they will die and have to respawn! Oh the humanity...

> They get busty chasing me, while my team promptly takes control of the payload and pushes the cart back in the absence of the enemy team. It's a sacrifice I'm willing to pay.

Wonderful. Another valid strategy for attacking/defending the payload.

Payload is a rather symmetric gamemode, mind you. So moments after you clear the payload with your airstrikes, some other smartass Yoyo on the enemy team will do the exact same to your team, using those same airstrikes. Which is perfectly fair.

> Yes, I suggest you try actual in-game testing, hence we can learn how effective the weapon is in game, and just exactly how similar it is to frag.

Would love to have that option, but alas it's not available. I would need to become a dev of this game to be able to implement and test abilities on staging server

**Comment by ghost** [comment]
>  How is it overpowered? You have 5 seconds to leave the area that is about to be blasted. This airstrike ability would literally be less powerful than frag grenades. It's not supposed to get kills, and it would rarely get any

Actually, you are correct, it isn't overpowered at all, it's the opposite. 

> Instead i could call an airstrike on them, forcing them to move in a specific way, perhaps forcing them to leave their cover and walk into my line of fire.

And why can't they just keep hugging the wall, hiding behind cover, and simply moving away from the range of the airstrike? They don't need to worry about moving, they barely have to move if they want to be out of the airstrike's range. The flaw of this device, is that **its range is just too small.** 

> The point is to make them move, yes. That is the entire point of this suggestion. An ability that lets players flush out other players from behind cover.

Yes, you are correct about me being wrong about it being overpowered. 
But no, it won't do a good job of flushing out players, because its just like a frag, but worse. 

> So increase the range then. As i said multiple times, the range is an educated guess. It would need to be updated after further testing.

Yes, that is the exact issue. **The range is so small, if it were implemented, it would just be a slow, and bad grenade.**
The difference between this, and a frag, is that a frag would give you less time to react. However, if you throw a frag, that takes a whole 10 seconds to explode, then its stupid, because it lacks the range to actually scare players far away from a zone. They would, move, what, 6 blocks away from the grenade? And they would be perfectly fine, all which would take them way less than 10 seconds, and just enough time to plan their next move. 
If you want to make a zone restricting weapon, then give it the range, as well as the cooldown to allow it to work properly. If I see sufficient range for the device, then I will glady agree with you, that this is a great device to add. 

> Then give it the necessities...

Yes. 

> The airstrike is strictly weaker than the frag grenade, so i fail to see why you're complaining.

Yeah, you're right, I shouldn't be complaining about it being overpowered, its clearly the opposite. 

> Oh no, Yoyo is threatening us with the idea of him playing the game the way it's supposed to be played! If you kill them, they will die and have to respawn! Oh the humanity...

...

> So moments after you clear the payload with your airstrikes, some other smartass Yoyo on the enemy team will do the exact same to your team, using those same airstrikes. Which is perfectly fair.

Lmfao either way, the cart will run backwards if not controlled by the enemy team, because they're too busy chasing me.

> Would love to have that option, but alas it's not available. I would need to become a dev of this game to be able to implement and test abilities on staging server

Shame.

**Comment by deferred-impact** [comment]
If you think i'm against increasing the range of this thing, you're wildly mistaken. I kept the range tiny to make it a more agreeable suggestion. However personally i'd gladly give it enough range to cover half the screen, lmao.

**Comment by ghost** [comment]
lmfao. You decreased the range to get agreement, and got the opposite. For something that is specifically designed to restrict an area, **I would give it a reasonably high range of damage,** but also make it just as slow.
